agent/consul/state/catalog.go

```diff
@@ -3229,3 +3241,56 @@ func cleanupMeshTopology(tx *txn, idx uint64, service *structs.ServiceNode) erro
 	}
 	return nil
 }
+
+func insertGatewayServiceTopologyMapping(tx *txn, idx uint64, gs *structs.GatewayService) error {
+	// Only ingress gateways are standalone items in the mesh topology viz
+	if gs.GatewayKind != structs.ServiceKindIngressGateway || gs.Service.Name == structs.WildcardSpecifier {
```

Comment:
Wildcard ingress is not included here because it would just point to everything?

Reply:
Right, we're already adding non-wildcard entries for the services that exist so I'm hooking into that.